### ISSUE TITLE
rx-v2020 not working after a few days

### ISSUE THREAD

**Issue by karia001** [wuub/rxv#68]
i have the problem that after a while the receiver is not reporting is state to HA anymore, reveiver stays pingable and is working fine from the app on the phone, only HA does not work anymore
is see in the logs the following
```
2020-12-26 21:06:56 ERROR (MainThread) [homeassistant.helpers.entity] Update for media_player.yamaha_receiver fails
  Traceback (most recent call last):
  File "/usr/local/lib/python3.8/xml/etree/ElementTree.py", line 1700, in close
  self.parser.Parse("", 1) # end of data
  xml.parsers.expat.ExpatError: no element found: line 1, column 0
  During handling of the above exception, another exception occurred:
  Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 278, in async_update_ha_state
  await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 474, in async_device_update
  raise exc
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
  result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/yamaha/media_player.py", line 195, in update
  self._play_status = self.receiver.play_status()
  File "/usr/local/lib/python3.8/site-packages/rxv/rxv.py", line 467, in play_status
  src_name = self._src_name(self.input)
  File "/usr/local/lib/python3.8/site-packages/rxv/rxv.py", line 254, in input
  response = self._request('GET', request_text)
  File "/usr/local/lib/python3.8/site-packages/rxv/rxv.py", line 154, in _request
  response = cElementTree.XML(res.content)
  File "/usr/local/lib/python3.8/site-packages/defusedxml/common.py", line 132, in fromstring
  return parser.close()
  File "/usr/local/lib/python3.8/xml/etree/ElementTree.py", line 1702, in close
  self._raiseerror(v)
  File "/usr/local/lib/python3.8/xml/etree/ElementTree.py", line 1602, in _raiseerror
  raise err
  File "<string>", line None
  xml.etree.ElementTree.ParseError: no element found: line 1, column 0
```
after restarting HA it starts working again

**Comment by RestOp** [comment]
I confirm, same thing...

**Comment by RestOp** [comment]
Here are some more infos about issue
https://community.home-assistant.io/t/yamaha-receiver-official-intergration-works-1-2-times-and-then-looses-control-update-problem/245121?u=thisio

**Comment by rdgerken** [comment]
I have the same issue with RX-A3080.  I'm willing to help troubleshoot!

**Comment by rdgerken** [comment]
I'm not sure the root cause yet... but it seems to be an invalid response (like an empty response) from the receiver, and perhaps an unhandled expat exception that is causing it to disconnect unrecovered from HA.  I see the try block at line 147 only handles a parse error, not an expat error.  Could this be the issue?  Do we need to have an error handler for all other exceptions in this try block (like what was done at line 120)?

**Comment by wuub** [comment]
I just reconnected my own Yamaha after almost 8 months.
I'll keep an eye out to see if this happens on my end

**Comment by rdgerken** [comment]
I did some testing on this to see if adding additional error handling would work, and I do not think it does.  To reproduce the issue, I had everything connected, and it worked as expected.  I manually broke the network connection (by unplugging upstream switch) - this caused my error handler that I added to catch a "Connection aborted" error.  After this, I plugged the AVR back into the network, waited for it to show back up on the network, and then attempted to control it using HA.  At this point, I get "Invalid XML returned for request <YAMAHA_AV cmd="GET"><Main_Zone><Input><Input_Sel>GetParam</Input_Sel></Input></Main_Zone></YAMAHA_AV>: b''  - I have no idea what this "b" is that is being returned.  But after that, HA then repeatedly reports that the update for media_player fails... and HA will never recover the connection - and as others have reported, it requires HA to be restarted to get going again.  So, it appears that when the connection is broken, upon trying to update after a connection break, it throws some "failed to update errors" and the connection never gets re-established.

Sorry, I know this isn't super technical, but hopefully it will help.  I do have some wireshark captures, but what I found is that when we get into this state, I never actually see the requests coming into the receiver - so its as if the requests are never actually making it out of HA (presumably because the connection is never getting re-established).

Thank you for your help!

**Comment by wuub** [comment]
Thank you @rdgerken this could actually help in tracking this down.
After reading what you said I have my suspicion https://github.com/wuub/rxv/blob/main/rxv/rxv.py#L106

**Comment by esev** [comment]
@rdgerken by any chance, did the IP address of the receiver change when it was plugged back into the network?

**Comment by esev** [comment]
It doesn't look like the `self._session.post` call inside of `_request` is setting any timeout. I wonder if the exception is a side effect of something else happening. Could it be that the session http request pool is full and is not accepting any more requests?

I wonder if adding a `timeout=60` argument to `self._session.post` would clear up the problem. Can someone who is having the issue test that?

**Comment by rdgerken** [comment]
The IP address is statically assigned on my receiver, so it should not have changed. 

Thanks!

**Comment by esev** [comment]
Are folks seeing this error message in the HomeAssistant logs also, or just the exception error message?

`Updating yamaha media_player took longer than the scheduled update interval 0:00:10`

I can fairly reliably get my HA instance to stop working with the receiver now. When it gets into that state I see the error message above every 10 seconds or so.

**Comment by rdgerken** [comment]
Yes, I see this repeatedly and infinitely until I restart HA - after I get the invalid XML returned message in my earlier post.

**Comment by esev** [comment]
Thanks for confirming. Just wanted to make sure I was solving the same issue and not a new one. Adding the timeout=60 parameter fixes the issue for me. I'll send a PR shortly.

**Comment by rdgerken** [comment]
OK great!  I'll give that a shot once I see the change.  Thanks very much for helping with this!

**Comment by esev** [comment]
You're welcome. Thanks for all the debugging information! That's what lead me toward the fix in #76.
